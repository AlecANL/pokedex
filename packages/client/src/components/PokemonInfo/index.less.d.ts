// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'image-container': string;
  'information-container': string;
  'pokemon-container': string;
  'title': string;
  'type': string;
}
declare var cssExports: CssExports;
export = cssExports;
