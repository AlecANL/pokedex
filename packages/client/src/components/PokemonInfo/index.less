.pokemon-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  .image-container {
    max-width: 180px;
    max-height: 180px;
    margin-top: -90px;

    img {
      width: 100%;
    }
  }
  .title {
    margin-top: -10px;
    span {
      font-size: 21px;
    }
  }

  .title:not(:last-child):after {
    content: ' / ';
  }

  .type {
    margin-top: 8px;
    display: flex;
    > div {
      padding: 5px;
    }
  }

  .information-container {
    display: flex;
    width: 80%;
    text-align: center;
    margin-top: 7px;
    justify-content: center;
    padding-bottom: 5px;

    > div {
      padding-left: 15px;
      padding-right: 15px;
      padding-top: 5px;
      padding-bottom: 5px;
      flex: 1;

      > div {
        padding: 2px;
        .title {
          font-weight: 600;
          font-size: 15px;
        }
      }
    }

    > div:not(:last-child) {
      border-right: rgba(112, 112, 112, 0.42) solid thin;
    }
  }
}
