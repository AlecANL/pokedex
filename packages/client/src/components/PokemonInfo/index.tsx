import React from 'react';
import style from './index.less';
import Panel from '../Panel';
import { Tooltip } from 'antd';
import PokemonType from '@/components/PokemonType';
import { formatMessage } from 'umi-plugin-react/locale';

interface IPokemonInfoProps {
  item?: IEdge;
}

const PokemonInfo: React.FC<IPokemonInfoProps> = props => {
  const { item } = props;
  return (
    <Panel
      content={
        <div className={style['pokemon-container']}>
          <div className={style['image-container']}>
            <img
              src={`https://assets.pokemon.com/assets/cms2/img/pokedex/detail/${item?.node.id}.png`}
            />
          </div>

          <div className={style.title}>
            <span>{item?.node.name}</span>
          </div>

          <div className={style.type}>
            {item?.node.types.map(type => {
              return (
                <Tooltip placement="bottom" title={type}>
                  <div>
                    <PokemonType size={38} icon={type} />
                  </div>
                </Tooltip>
              );
            })}
          </div>

          <div className={style['information-container']}>
            <div>
              <div>
                {item?.node.types.map(type => (
                  <span className={style['title']}>{type}</span>
                ))}
              </div>
              <div>
                <span>{formatMessage({ id: 'pages.type' })}</span>
              </div>
            </div>
            <div>
              <div>
                <span className={style['title']}>{item?.node.weight.maximum}</span>
              </div>
              <div>
                <span>{formatMessage({ id: 'pages.weight' })}</span>
              </div>
            </div>
            <div>
              <div>
                <span className={style['title']}>{item?.node.height.maximum}</span>
              </div>
              <div>
                <span>{formatMessage({ id: 'pages.height' })}</span>
              </div>
            </div>
          </div>
        </div>
      }
    />
  );
};

export default PokemonInfo;
