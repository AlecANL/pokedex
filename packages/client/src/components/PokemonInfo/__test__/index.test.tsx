import React from 'react';
import renderer from 'react-test-renderer';
import PokemonInfo from '..';

jest.mock('umi-plugin-react/locale');

describe('Components/PokemonInfo', () => {
  it('Get snapshot', () => {
    const component = renderer.create(<PokemonInfo item={undefined} />);
    let snap = component.toJSON();
    expect(snap).toMatchSnapshot();
  });
});
