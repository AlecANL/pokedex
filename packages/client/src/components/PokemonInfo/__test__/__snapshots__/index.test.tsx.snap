// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/PokemonInfo Get snapshot 1`] = `
<div
  className="panel-component"
>
  <div>
    <div
      className="pokemon-container"
    >
      <div
        className="image-container"
      >
        <img
          src="https://assets.pokemon.com/assets/cms2/img/pokedex/detail/undefined.png"
        />
      </div>
      <div
        className="title"
      >
        <span />
      </div>
      <div
        className="type"
      />
      <div
        className="information-container"
      >
        <div>
          <div />
          <div>
            <span />
          </div>
        </div>
        <div>
          <div>
            <span
              className="title"
            />
          </div>
          <div>
            <span />
          </div>
        </div>
        <div>
          <div>
            <span
              className="title"
            />
          </div>
          <div>
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
